vue-app/src/pages/TablePage.vue

```diff
@@ -0,0 +1,242 @@
+<template>
+  <div class="page-wrapper">
+    <div class="page-content">
+      <div class="v-btn-cont">
+        <button @click="addNRecords(RECORDS_TO_CREATE)">
+          Add {{ RECORDS_TO_CREATE }}
+        </button>
+        <button @click="deleteNRecords(RECORDS_TO_DELETE)">
+          Delete {{ RECORDS_TO_DELETE }}
+        </button>
+        <button @click="deleteEveryNthRecord(NTH_TO_DELETE)">
+          Delete {{ NTH_TO_DELETE }}th
+        </button>
+        <button @click="updateNthRow(NTH_TO_UPDATE)">
+          Update {{ NTH_TO_UPDATE }}th
+        </button>
+        <button @click="replaceAllRows">Replace all</button>
+        <button @click="swapRows">Swap</button>
+        <button @click="clearRows">Clear all</button>
+        <span>rows: {{ rowCount }}</span>
+      </div>
+
+      <div class="table-container">
+        <div class="table">
+          <div class="table-header">
+            <div>Id</div>
+            <div>Name</div>
+            <div>Level</div>
+          </div>
+          <table-row
+            class="table-row"
+            :key="tuple.id"
+            v-for="tuple in tableContent"
+            :tuple="tuple"
+          >
+          </table-row>
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { computed, ComputedRef, onBeforeMount, ref } from "vue";
+import { namesTable } from "../static/RandomDataTables";
+import { numberTable } from "../static/RandomDataTables";
+import TableRow from "../components/UI/TableRow.vue";
+import { TableRowData } from "../models/PerfTestArrayRow";
+
+const RECORDS_TO_CREATE: number = 3000;
+const RECORDS_TO_DELETE: number = 1000;
+const NTH_TO_DELETE: number = 2;
+const NTH_TO_UPDATE: number = 2;
+
+const tableContent = ref<TableRowData[]>([]);
+const rowCount: ComputedRef<number> = computed(() => {
+  return tableContent.value.length;
+});
+
+let key: number = 0;
+
+onBeforeMount(generateArray);
+
+function addNRecords(n: number) {
+  let nameIndex: number;
+  let levelIndex: number;
+
+  const tmpArray: TableRowData[] = [];
+
+  for (let i = 0; i < n; i++) {
+    nameIndex = Math.floor(Math.random() * namesTable.length);
+    levelIndex = Math.floor(Math.random() * numberTable.length);
+
+    tmpArray.unshift({
+      id: key++,
+      name: namesTable[nameIndex],
+      level: numberTable[levelIndex],
+    });
+  }
+
+  tableContent.value.unshift(...tmpArray);
+}
+
+function deleteNRecords(n: number) {
+  tableContent.value.splice(0, n);
+}
+
+function deleteEveryNthRecord(n: number) {
+  const tmpArray: TableRowData[] = [...tableContent.value];
+
+  for (let i = 0; i < tmpArray.length; i += n) {
+    tmpArray.splice(i--, 1);
+  }
+
+  tableContent.value = tmpArray;
+}
+
+function updateNthRow(n: number) {
+  const tmpArray: TableRowData[] = [...tableContent.value];
+
+  for (let i = 0; i < tableContent.value.length; i += n) {
```

Comment:
It would be better to use rowContent variable

Reply:
could you explain why and how?

Reply:
I meant rowCount variable, just a styling purpose.
when you already have a variable with the value, you can use it instead of drilling into object